core/src/main/java/org/infinispan/util/concurrent/locks/OwnableReentrantLock.java

```diff
@@ -40,6 +42,7 @@ protected final Object currentRequestor() {
    }
 
    private void setCurrentRequestor(Object requestor) {
+      assert requestorOnStack.get() == null;
```

Comment:
But assert does hurt, if it slows down the test suite :)

I'd remove the assert altogether, and maybe add a TL leak test for `OwnableReentrantLock`. The TL is only used internally, so there's no point to check it every single time.